runtime/lua/vim/lsp/diagnostic.lua

```diff
@@ -253,6 +253,7 @@ end
 ---                        If nil, diagnostics of all clients are included.
 ---@return table with diagnostics grouped by bufnr (bufnr: Diagnostic[])
 function M.get_all(client_id)
+  vim.api.nvim_echo([['vim.lsp.diagnostic.get_all is deprecated. Use vim.diagnostic.get instead', 'WarningMsg']], true, {})
```

Comment:
I like @clason's suggestion to instead refer users to `:h deprecated`, which already contains the replacements for each deprecated function, rather than suggesting the replacement for each function individually.

```suggestion
  vim.api.nvim_echo([['vim.lsp.diagnostic.get_all is deprecated. See :h deprecated', 'WarningMsg']], true, {})
```

We can workshop the actual wording/phrasing (our resident wordsmith @clason likely has the *mot juste*).